lib/src/events/ComponentEventsObserver.ts

```diff
@@ -38,11 +38,9 @@ export class ComponentEventsObserver {
     this.nativeEventsReceiver.registerPreviewCompletedListener(this.notifyPreviewCompleted);
   }
 
-  public bindComponent(component: React.Component<any>): EventSubscription {
-    const componentId = component.props.componentId;
-    if (!_.isString(componentId)) {
-      throw new Error(`bindComponent expects a component with a componentId in props`);
-    }
+  public bindComponent(component: React.Component<any>, componentIdArg?: string): EventSubscription {
+    const componentId = componentIdArg ? componentIdArg : component.props.componentId;
+
     if (_.isNil(this.listeners[componentId])) {
```

Comment:
Also we don't need to use lodash here you can do it like this `if (!this.listeners[componentId]) {` 👍